libcnb-data/src/build_plan.rs

```diff
@@ -119,4 +154,23 @@ mod tests {
 
         assert!(toml::to_string(&build_plan).is_ok());
     }
+
+    #[test]
+    fn it_serializes_metadata() {
+        #[derive(Serialize)]
+        struct Metadata {
+            foo: String,
+        }
+
+        let mut require = Require::new("foo");
+        let metadata = Metadata {
+            foo: String::from("bar"),
+        };
+        let result = require.metadata(metadata);
+        assert!(result.is_ok());
+        assert_eq!(
+            require.metadata.get("foo"),
+            Some(&toml::Value::String(String::from("bar")))
+        );
```

Comment:
Can we streamline the assert here so it works the same as in `it_deserializes_metadata`?

Reply:
@Malax what were you thinking here? Unlike the other test, I can derive traits on the `Metadata` struct b/c that's what I'm checking against, but I have a `Table` here.

Reply:
I tried adding `Eq` to `Require` but sadly it's not implemented on `toml::map::Map`. :(

https://docs.rs/toml/latest/toml/map/struct.Map.html